### PR TITLE
Fix golint failures in pkg/registry/autoscaling & pkg/registry/batch

### PR DESCRIPTION
**Pull request by CaoDonghui123**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix golint failures :
pkg/registry/autoscaling/rest
pkg/registry/batch/job
pkg/registry/batch/rest

**Which issue(s) this PR fixes**:

Part of #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by CaoDonghui123** [comment]
ping @hongchaodeng 
Could you review this PR?

File: pkg/registry/autoscaling/rest/storage_autoscaling.go [hunk 1]
```diff
@@ -29,39 +29,41 @@ import (
...
 	// TODO refactor the plumbing to provide the information in the APIGroupInfo
 
 	if apiResourceConfigSource.VersionEnabled(autoscalingapiv2beta2.SchemeGroupVersion) {
-		if storageMap, err := p.v2beta2Storage(apiResourceConfigSource, restOptionsGetter); err != nil {
-			return genericapiserver.APIGroupInfo{}, false, err
-		} else {
+		if storageMap, err := p.v2beta2Storage(apiResourceConfigSource, restOptionsGetter); err == nil {
 			apiGroupInfo.VersionedResourcesStorageMap[autoscalingapiv2beta2.SchemeGroupVersion.Version] = storageMap
+		} else {
+			return genericapiserver.APIGroupInfo{}, false, err
```
Comment (caesarxuchao) [inline pkg/registry/autoscaling/rest/storage_autoscaling.go:45]: The style convention returning on error first to reduce indention on the non-error case:
```golang
if storageMap, err := p.v2beta2Storage(apiResourceConfigSource, restOptionsGetter); err != nil {
  return genericapiserver.APIGroupInfo{}, false, err
}
apiGroupInfo.VersionedResourcesStorageMap[autoscalingapiv2beta2.SchemeGroupVersion.Version] = storageMap
```

Can you use the above pattern throughout this PR?

File: pkg/registry/autoscaling/rest/storage_autoscaling.go [hunk 2]
```diff
@@ -100,6 +102,7 @@ func (p RESTStorageProvider) v2beta2Storage(apiResourceConfigSource serverstorag
 	return storage, err
 }
 
-func (p RESTStorageProvider) GroupName() string {
+// GroupName is the group name use in this package
```
Comment (caesarxuchao) [inline pkg/registry/autoscaling/rest/storage_autoscaling.go:105]: How about "GroupName returns the canonical group name of the autoscaling group"?

File: pkg/registry/autoscaling/rest/storage_autoscaling.go [hunk 3]
```diff
@@ -29,39 +29,41 @@ import (
 	horizontalpodautoscalerstore "k8s.io/kubernetes/pkg/registry/autoscaling/horizontalpodautoscaler/storage"
 )
 
-type RESTStorageProvider struct{}
+// StorageProvider is a REST storage provider for storage_autoscaling
```
Comment (caesarxuchao) [inline pkg/registry/autoscaling/rest/storage_autoscaling.go:32]: How about "StorageProvider is a REST storage provider for autoscaling API"?

**Comment by CaoDonghui123** [comment]
@caesarxuchao Thank you so much. I fixed it

File: pkg/registry/batch/job/strategy.go [hunk 4]
```diff
@@ -185,6 +185,7 @@ type jobStatusStrategy struct {
 	jobStrategy
 }
 
+// StatusStrategy is the default logic invoked when updating object status
```
Comment (caesarxuchao) [inline pkg/registry/batch/job/strategy.go:188]: nit: please add a "." at the end.

**Review by caesarxuchao — COMMENTED** [review]
A nit. Otherwise lgtm.

File: pkg/registry/batch/job/strategy.go (hunk 4 above)
Comment (CaoDonghui123) [reply pkg/registry/batch/job/strategy.go]: > nit: please add a "." at the end.

I fixed it

File: pkg/registry/batch/rest/storage_batch.go [hunk 5]
```diff
@@ -30,39 +30,41 @@ import (
 	jobstore "k8s.io/kubernetes/pkg/registry/batch/job/storage"
 )
 
-type RESTStorageProvider struct{}
+// StorageProvider is a REST storage provider for bath API.
```
Comment (caesarxuchao) [inline pkg/registry/batch/rest/storage_batch.go:33]: s/bath/batch.

also, it should be "the batch API".

File: pkg/registry/autoscaling/rest/storage_autoscaling.go [hunk 6]
```diff
@@ -29,39 +29,41 @@ import (
 	horizontalpodautoscalerstore "k8s.io/kubernetes/pkg/registry/autoscaling/horizontalpodautoscaler/storage"
 )
 
-type RESTStorageProvider struct{}
+// StorageProvider is a REST storage provider for autoscaling API.
```
Comment (caesarxuchao) [inline pkg/registry/autoscaling/rest/storage_autoscaling.go:32]: nit: "for the autoscaling API"

Comment (CaoDonghui123) [reply pkg/registry/autoscaling/rest/storage_autoscaling.go]: thanks for your suggests

**Pull request closed without merging**